src/Randomizer.Data/WorldData/Item.cs

```diff
@@ -366,6 +371,12 @@ public static List<Item> CreateJunkPool(World world)
             itemPool.AddRange(Copies(15, () => new Item(ItemType.Super, world)));
             itemPool.AddRange(Copies(8, () => new Item(ItemType.PowerBomb, world)));
 
+            // If no quarter magic, add an extra 20 rupees back
+            if (!world.Config.CasPatches.ExpandedItems)
```

Comment:
Maybe we can change this a little bit to be more flexible, check how many items we expect in the pool, and fill up the remaining items with big 20s, rather than putting in logic like this.

Reply:
The different item pool generation functions were all static methods that didn't really have any good way of communicating, so I went and created a "WorldItemPools" class to generate all of the pools for a world. Using that, I'm able to tally up the number of items vs the number of locations and fill in the gaps with twenty rupees.